build/ninja-rules-cpp.sh

```diff
@@ -289,7 +289,11 @@ link() {
 
   # IMPORTANT: Flags like -ltcmalloc have to come AFTER objects!  Weird but
   # true.
-  $prefix "$cxx" -o "$out" "$@" $link_flags
+  local cmdline=($prefix "$cxx" -o "$out" "$@" $link_flags)
```

Comment:
This file unfortunately can't use arrays because it's sourced by the end user build, `_build/oils.sh` , which runs under `#!/bin/sh`

See the comments the analogous place in compile_one

So I guess copy and paste is the only real solution ...